code/modules/mob/living/carbon/give.dm

```diff
@@ -1,22 +1,23 @@
 /mob/living/carbon/verb/give()
 	set category = "IC"
 	set name = "Give"
-	set src in view(1)
+	set src in oview(1) //Cannot handle giving shit to mobs on your own tile, but it's a small, small loss
+
 	give_item(usr)
 
 /mob/living/carbon/proc/give_item(mob/living/carbon/user)
 
 	if(src.stat == 2 || user.stat == 2 || src.client == null)
 		return
-	if(src == user)
-		user << "\red I feel stupider, suddenly."
+	if(src == user) //Shouldn't happen
+		user << "<span class='warning'>You feel stupider, suddenly.</span>"
 		return
 	var/obj/item/I
 	if(!user.hand && user.r_hand == null)
-		user << "\red You don't have anything in your right hand to give to [src.name]"
+		user << "<span class='warning'>You don't have anything in your right hand to give to [src]</span>"
 		return
 	if(user.hand && user.l_hand == null)
```

Comment:
You should compress this check.
`if(user.get_active_hand() == null)`
`user << "You don't have anything in your [user.hand ? "left hand" : "right hand"] to give to [src]."`